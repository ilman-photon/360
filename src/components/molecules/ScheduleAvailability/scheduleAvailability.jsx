import Box from "@mui/material/Box";
import React, { useEffect, useState } from "react";
import { viewAllAvailabilityLinkUI } from "../WeekAvailability/WeekAvailability";
import { buttonSchedule } from "../DayAvailability/DayAvailability";
import { parseScheduleDataDay } from "../../../utils/appointment";

export const ScheduleAvailability = ({
  scheduleData = [],
  onClickViewAllAvailability = () => {
    // This is intentional
  },
  OnDayClicked = () => {
    // This is intentional
  },
  currentDateIndex = 0,
  currentDate = "",
}) => {
  const [schedule, setSchedule] = useState([]);

  useEffect(() => {
    const scheduleParse = parseScheduleDataDay(scheduleData, currentDateIndex);
    if (scheduleParse) {
      setSchedule(scheduleParse);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [currentDateIndex]);

  function isScheduleAvailable() {
    return (
      schedule.length === 1 && schedule[0].indexOf("Next availability is") > -1
    );
  }

  function isValidDate(d) {
    return d instanceof Date && !isNaN(d);
  }
  const stringCurrentDate = isValidDate(currentDate)
    ? currentDate.toLocaleDateString()
    : "";
  return (
    <Box>
      <Box
        sx={{
          display: "grid",
          width: !isScheduleAvailable() ? "328px" : "100%",
          gridTemplateColumns: !isScheduleAvailable()
            ? "repeat(4, 1fr)"
            : "repeat(1, 1fr)",
          gap: "4px",
          justifyContent: "center",
          alignContent: "center",
          gridTemplateRows: "auto",
        }}
      >
        {schedule.map((option, idx) => {
          return buttonSchedule(
            option,
            idx,
            OnDayClicked,
            stringCurrentDate,
            true
          );
        })}
      </Box>
      {viewAllAvailabilityLinkUI({ onClickViewAllAvailability })}
    </Box>
  );
};
