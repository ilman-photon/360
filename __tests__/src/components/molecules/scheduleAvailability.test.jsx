import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ScheduleAvailability } from "../../../../src/components/molecules/ScheduleAvailability/scheduleAvailability";

describe("ScheduleAvailability Components", () => {
  let container;
  beforeEach(() => {
    container = render(<ScheduleAvailability scheduleData={[
      {
        date: "2022-09-19",
        list: [
          {
            time: "11:30am",
            key: 12222,
          },
        ],
      },
      {
        date: "2022-09-20",
        list: [
          {
            time: "08:00am",
            key: 12223,
          },
          {
            time: "10:30am",
            key: 12224,
          },
          {
            time: "11:00am",
            key: 12225,
          },
          {
            time: "12:00pm",
            key: 12226,
          },
          {
            time: "13:00pm",
            key: 12227,
          },
          {
            time: "14:00pm",
            key: 12228,
          },
        ],
      },
      {
        date: "2022-09-21",
        list: [
          {
            time: "08:30am",
            key: 12229,
          },
          {
            time: "10:30am",
            key: 12230,
          },
          {
            time: "11:30am",
            key: 12231,
          },
          {
            time: "12:00pm",
            key: 12232,
          },
          {
            time: "13:30pm",
            key: 12233,
          },
          {
            time: "14:30pm",
            key: 12234,
          },
          {
            time: "15:30pm",
            key: 12235,
          },
          {
            time: "16:30pm",
            key: 12236,
          },
          ,
        ],
      },
      {
        date: "2022-09-22",
        list: [
          {
            time: "09:30am",
            key: 12237,
          },
          {
            time: "11:00am",
            key: 12238,
          },
        ],
      },
      {
        date: "2022-09-23",
        list: [
          {
            time: "09:30am",
            key: 12239,
          },
        ],
      },
      {
        date: "2022-09-24",
        list: [
          {
            time: "09:30am",
            key: 12240,
          },
        ],
      },
    ]} currentDateIndex={0}/>)
  });

  it("ScheduleAvailability render", () => {
    expect(container).toMatchSnapshot();
  });
});
