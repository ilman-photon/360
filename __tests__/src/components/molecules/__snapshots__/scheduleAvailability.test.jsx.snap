// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleAvailability Components ScheduleAvailability render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-8rfnb2"
        >
          <div
            class="scheduleBtnWarpper MuiBox-root css-8atqhb"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleAvailBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
              data-testid="appointment_all_availability_timeslot_button"
              tabindex="0"
              type="button"
            >
              11:30am
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="linkWrapper MuiBox-root css-wdc560"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways linkAvailabelity css-1ps4owl-MuiTypography-root-MuiLink-root"
            data-testid="loc_viewAll"
          >
            View all availability
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-8rfnb2"
      >
        <div
          class="scheduleBtnWarpper MuiBox-root css-8atqhb"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleAvailBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            data-testid="appointment_all_availability_timeslot_button"
            tabindex="0"
            type="button"
          >
            11:30am
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="linkWrapper MuiBox-root css-wdc560"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways linkAvailabelity css-1ps4owl-MuiTypography-root-MuiLink-root"
          data-testid="loc_viewAll"
        >
          View all availability
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
